### PR TITLE
zfs: update to 2.2.0, add zfs-lts package

### PR DESCRIPTION
**Pull request by zdykstra**
#### Testing the changes
- I tested the changes in this PR: **YES**

Place-holder PR so that we can get this out the door once `2.2.0` officially lands.

### CODE REVIEW

**Comment by zdykstra** [comment]
Upstream fix for the broken test cmd at https://github.com/openzfs/zfs/pull/15130

**Comment by dkwo** [comment]
On the last coreutils update, this was reported: Locally I get a fail in tests/cp/sparse-2.sh, but I'm on ZFS...

**Comment by zdykstra** [comment]
Looks like this is a reiteration of https://github.com/openzfs/zfs/issues/14555 . I don't see any real closure to that issue. Does `ZFS 2.1.12` build against the same kernel?

**Comment by zdykstra** [comment]
Determining which course of action to follow is in @Vaelatern's court. Barring any additional input from him, I won't take any action to resolve this, for now.

**Comment by Vaelatern** [comment]
@r-ricci could we tell ZFS to lie about its license only on aarch64? Having read the justification for the GPL symbol limit (that it's meant for internal-only interfaces that could only have violated the GPL unless GPL), I think such a patch, on the user's machine, just before running DKMS, only on neon architectures, would be reasonable. This might be the right solution for now while upstream linux and zfs fight it out.

**Comment by Vaelatern** [comment]
Let's make the META file a conf file. Let's also leave it as default broken on aarch64. Let's finally consider documenting this on the void linux doc site... and leave an open issue in void-packages about zfs being broken on aarch64, so others may change the license and get a full speed ZFS themselves.

**Comment by zdykstra** [comment]
Bug report opened at https://github.com/void-linux/void-packages/issues/45552 and referenced in the template.  `/usr/src/zfs/zfs-${version}/META"` set to be a configuration file.

**Comment by zdykstra** [comment]
> Let's make the META file a conf file. Let's also leave it as default broken on aarch64. Let's finally consider documenting this on the void linux doc site... and leave an open issue in void-packages about zfs being broken on aarch64, so others may change the license and get a full speed ZFS themselves.

Does /usr/src/zfs/zfs-${version}/META need to be marked as a configuration file? Any future updates to ZFS will install under a different versioned directory.

**Comment by classabbyamp** [comment]
it would get overwritten on reinstalls of the same version (or revbumps)

File: srcpkgs/zfs/template [hunk 1]
```diff
@@ -10,19 +10,23 @@ hostmakedepends="dracut libtool pkg-config python3 automake"
...
+distfiles="https://github.com/openzfs/zfs/releases/download/zfs-${version}-rc3/zfs-${version}-rc3.tar.gz"
+checksum=276c911ab4aa24b5c261e1e1822f74ed824a424cacf3f39b14da1aa08ce1906c
 # dkms must be before initramfs-regenerate to build modules before images
 triggers="dkms initramfs-regenerate"
 dkms_modules="zfs ${version}"
 
 replaces="spl<=0.7.13_1" # Because SPL was merged into zfs in 0.8.0
 
+case "$XBPS_MACHINE" in
+	aarch64*) broken "NEON symbols are GPL only, see https://github.com/void-linux/void-packages/issues/45552"
```
[the remark this answers is not in the document]
Comment (Vaelatern) [reply srcpkgs/zfs/template:27]: I don't want it to be `broken`. Let the ZFS DKMS build fail on the target machine.

**Comment by zdykstra** [comment]
Based on what I'm reading and seeing, the ZFS 2.1.x branch is going to continue to receive fixes and back-ports. https://github.com/openzfs/zfs/tree/zfs-2.1.13-staging exists, and there's at least one open PR for cherry picked fixes going to that branch.

Do we want to consider offering a `zfs2.1` package that tracks the ZFS 2.1.x branch? That'd leave the 'zfs' package following the most recent release - and people could pick how much risk they want in their ZFS implementation.

**Comment by Vaelatern** [comment]
Is it officially "LTS" or just a special one-time support for 2.1?

**Comment by zdykstra** [comment]
> Is it officially "LTS" or just a special one-time support for 2.1?

It's officially an LTS release; https://github.com/openzfs/zfs/blob/master/RELEASES.md

**Comment by classabbyamp** [comment]
if we do it, maybe call it zfs-lts so we don't have to add/remove packages for later LTS?

**Comment by Vaelatern** [comment]
"New kernel versions released after the OpenZFS LTS release are not supported" concerns me. We already hold moving the `linux` version for ZFS, this might make moving `linux-lts` very bad considering the 2 year support window.

**Comment by zdykstra** [comment]
I'd say that `zfs-lts` shouldn't gate moving the `linux` kernel version. If you want to run an LTS ZFS, run a (supported) LTS kernel.

**Comment by ahesford** [comment]
General thoughts:
1. Should `zfs-lts` and its subpackages  provide or replace (or both) the equivalent non-LTS packages? This might support a smoother replacement experience for users.
2. I wonder if the META file should be a `conf_file` only for `aarch64`. Configuration files have some warts like the installation of original copies alongside their modified versions, so we can limit the damage to where it's needed.

**Comment by zdykstra** [comment]
> General thoughts:
> 
> 1. Should `zfs-lts` and its subpackages  provide or replace (or both) the equivalent non-LTS packages? This might support a smoother replacement experience for users.

Either/or is fine with me. 

> 2. I wonder if the META file should be a `conf_file` only for `aarch64`. Configuration files have some warts like the installation of original copies alongside their modified versions, so we can limit the damage to where it's needed.

That's probably a good call.

With ZFS 2.2.0 RC4 out, upstream is probably close to a full release. Thoughts on these two items, @Vaelatern ?

**Comment by Vaelatern** [comment]
I'm happy with lts providing regular zfs, so long as switching to main zfs is supported too. Really if you upgrade your pool to newest, it'll no longer be supported by LTS I think.

Also happy with conf_file only on aarch64, but I think it might look ugly or look unmaintainable. No problem IMO with the conf_file on every arch.

**Comment by zdykstra** [comment]
The lint stage is failing because I'm starting `zfs-lts` off with a revision of 3, to avoid transitioning backwards from zfs-2.2.12_2 in the event that people switch to `zfs-lts`.  Once upstream officially tags `2.2.0` I'll flip the relevant template and this should be ready to go.

**Comment by darkbasic** [comment]
I'm sorry for the stupid question but I'm not a void-linux user: I need to build zfsbootmenu via its [Dockerfile](https://github.com/zbm-dev/zfsbootmenu/blob/master/releng/docker/Dockerfile#L23) using zfs 2.2 otherwise I won't be able to boot into my system because of unsupported features (latest version of zfsbootmenu is 2.2 but apparently uses zfs 2.1). What's the easiest way to achieve this? Is there any compiled package triggered by the CI that I can download? If not how can I modify the [Dockerfile](https://github.com/zbm-dev/zfsbootmenu/blob/master/releng/docker/Dockerfile#L23) to compile zfs 2.2 from this PR? Thanks

**Comment by Vaelatern** [comment]
Let's let this sit for a day or two, just to let other people test first, then :+1:

**Comment by Vaelatern** [comment]
I'm very excited about corrective zfs receive

**Comment by ahesford** [comment]
I'm very excited about block cloning.

**Comment by classabbyamp** [comment]
I'm very excited about container support stuff

**Comment by darkbasic** [comment]
> I'm very excited about block cloning

That works very well, but be aware that you won't be able to clone between datasets if you use encryption.

> Let's let this sit for a day or two, just to let other people test first

I've been testing 2.2 for weeks and it's mostly rock solid. I've encountered a single data corrupting bug but they managed to reproduce it and temporarily disable the offending feature just in the time for the stable release.

**Comment by heppu** [comment]
Oh boy have I been waiting for this! Can confirm what @Vaelatern said; pool cannot be used with older zfs version if upgraded. Without thinking it about it I upgraded my pools from arch linux which had zfs-git installed and since then I have been manually installing zfs 2.2-rc versions on my void. 😅

**Comment by Vaelatern** [comment]
@zdykstra I can't IRC right now, but :+1: for this.

File: srcpkgs/zfs-lts/template [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+version=2.1.13
+revision=1
+build_style=gnu-configure
+configure_args="--with-config=user --with-mounthelperdir=/usr/bin
+ --with-udevdir=/usr/lib/udev --with-udevruledir=/usr/lib/udev/rules.d
+ --with-dracutdir=/usr/lib/dracut --with-tirpc"
+hostmakedepends="dracut libtool pkg-config python3 automake"
+makedepends="pam-devel zlib-devel libuuid-devel libblkid-devel libtirpc-devel attr-devel"
+depends="dkms>=2.2.0.3_9 perl python3 xbps-triggers>=0.121_1
+ libzfs>=${version}_${revision}"
```
Comment (classabbyamp) [inline srcpkgs/zfs-lts/template:12]: libzfs-lts?

Comment (zdykstra) [reply srcpkgs/zfs-lts/template]: Good eye

**Pull request merged**